### PR TITLE
shell.nix: reorder, tidy up and remove duplicates

### PR DESCRIPTION
**Pull request by orangecms**

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -4,55 +4,68 @@ let
...
       # Copied from another Electron application. Probably doesn't need everything
       bash
       coreutils
       zlib
       stdenv.cc.cc
-      xorg.libXext
-      xorg.libX11
+      xorg.libXft
+
+      # Copied from atom
```
Comment (JohnAZoidberg) [inline shell.nix:25]:
```suggestion
      # Copied from atom (because it's also an Electron app)
```

File: shell.nix [hunk 2]
```diff
@@ -4,55 +4,68 @@ let
...
+      mesa # for libgbm
 
       # Manually added
       xorg.libxshmfence
-      nss
-      nspr
-      atk
-      at-spi2-atk
-      dbus
-      libdrm
```
Comment (JohnAZoidberg) [inline shell.nix:38]: I put them here under this comment so we know these are strictly required.
The ones at the top, copied from atom are probably not all required. So those we have to double-check, these here we don't.

**Comment by JohnAZoidberg** [comment]
Still works

File: shell.nix (hunk 2 above)
Comment (orangecms) [reply shell.nix:38]: Makes sense, I'll move those again, thank you!

**Comment by orangecms** [comment]
I've dropped shell.nix for now, having no maintainer for it.

**Pull request closed without merging**